src/PrettierESLintLinter.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Copyright 2018 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Lints JavaScript and JSX files using Prettier & Eslint auto-fix
+ */
+final class PrettierESLintLinter extends ArcanistExternalLinter {
+  private $cwd = '';
+  private $flags = array();
+
+  public function getInfoName() {
+    return 'PrettierESLint';
+  }
+
+  public function getInfoURI() {
+    return 'https://github.com/prettier/prettier-eslint-cli';
+  }
+
+  public function getInfoDescription() {
+    return pht('A combo Prettier formatter & Eslint auto-fix linter');
+  }
+
+  public function getLinterName() {
+    return 'PRETTIERESLINT';
+  }
+
+  public function getLinterConfigurationName() {
+    return 'prettier-eslint';
+  }
+
+  public function shouldUseInterpreter() {
+    return true;
+  }
+
+  public function getDefaultInterpreter() {
+    list($err, $stdout, $stderr) = exec_manual('node -v');
+    preg_match('/^v([^\.]+)\..*$/', $stdout, $m);
+    if (array_key_exists(1, $m) && (int)$m[1] < 6) {
+      // Only used for node < 6
+      return __DIR__ . '/node4_proxy';
+    }
+    // Also used if node doesn't exist, causing an install instructions error
+    return 'node';
+  }
+
+  public function getDefaultBinary() {
+    list($err, $stdout, $stderr) = exec_manual('yarn -s --cwd %s which prettier-eslint', $this->getProjectRoot() . '/' . $this->cwd);
+    $binaryPath = strtok($stdout, "\n");
+    if ($binaryPath == "") {
```

Comment:
`if (empty($binaryPath)) {` is a little more idiomatic.

Reply:
Fixed.